### PR TITLE
py3 changes, missing statements, normalisation of paths.

### PR DESCRIPTION
**Pull request by d-fens**
This adds in python3 compatibility when used pyldap which is compatible with python-ldap with no changes required.

Changed path join operations to avoid string concatenation so as to be more portable, however it is heavily based on a UNIX system layout, so it's mostly irrelevant.

Changes to be more inline with PEP8.

Missing statement in password generation. Still requires logic to be added there.

Added in dependencies.

Added gitignore for python.

### CODE REVIEW

**Review by lil-cain — APPROVED** [review]
Please ignore, I fail at github

**Review by butlerx — APPROVED** [review]

File: useradm/rbconfig.py [hunk 1]
```diff
@@ -115,26 +115,26 @@
...
+        'founders':     'RedBrick founder',
+        'member':       'Normal member',
+        'associat':     'Graduate/associate member',
+        'staff':        'DCU staff member',
+        'society':      'DCU society',
+        'club':         'DCU club',
+        'projects':     'RedBrick/DCU/Course project account',
+        'guest':        'Guest account',
+        'intersoc':     'Account for society from another college',
+        'committe':     'Committee member or a position account',
```
Comment (VoyTechnology) [inline useradm/rbconfig.py:127]: Not exactly part of what this change is about, but should it not be `committee`?

Comment (d-fens) [reply useradm/rbconfig.py:127]: Correct, however based on years old NIS constraints on group names that probably don't apply anymore, this is called 'committe' so as to fall into the 8 char length and this has been the case for some time on Redbrick.

**Review by VoyTechnology — APPROVED** [review]
Tried running the basic thing without going too in depth and glanced over the code. Works, 

LGTM

**Pull request merged**